src/main/java/uk/ac/cam/cl/dtg/util/mappers/MiscMapper.java

```diff
@@ -0,0 +1,69 @@
+package uk.ac.cam.cl.dtg.util.mappers;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.factory.Mappers;
+import uk.ac.cam.cl.dtg.isaac.dos.AssignmentDO;
+import uk.ac.cam.cl.dtg.isaac.dos.GameboardContentDescriptor;
+import uk.ac.cam.cl.dtg.isaac.dos.GameboardDO;
+import uk.ac.cam.cl.dtg.isaac.dos.QuizAssignmentDO;
+import uk.ac.cam.cl.dtg.isaac.dos.QuizAttemptDO;
+import uk.ac.cam.cl.dtg.isaac.dto.AssignmentDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.GameboardDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.GameboardItem;
+import uk.ac.cam.cl.dtg.isaac.dto.QuizAssignmentDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.QuizAttemptDTO;
+import uk.ac.cam.cl.dtg.isaac.dto.ResultsWrapper;
+
+@Mapper
+public interface MiscMapper {
```

Comment:
This mapper looks OK right now, it's small, and it only has a handful of methods. However, given its name, this has the potential to grow out of control. If we were to break this down into more specific mappers, how would those mappers look like? It's OK if some of them only include a single method. If you could let me know the names of those new mappers, and the name of methods they would contain, then we can make a decision and possibly refactor this.

Reply:
I've grouped methods in the larger mappers based on classes, since the subclass mappings need to accessible to their parents, along with occasional other objects used in the class' properties and not much else. These might want reorganising as well if we're looking to split the MiscMapper... 🤔 
If we similarly split the MiscMapper according to classes, we would have additional interfaces for Gameboard, Assignment, QuizAssignment and QuizAttempt, each with a pair of DO <-> DTO mappings. ResultsWrapper would presumably add a fifth interface, while the `GameboardContentDescriptor <-> GameboardItem` mapping could either become a sixth or be folded under the Gameboard interface where it is currently used. So 5-6 there.
If we clean up any loose mappings from the larger interfaces as well, that could also add one for QuizSummary and ContentSummary, as well as Address and/or Location if we decide that needs splitting out... I'm not Maybe GroupMembership and Group from the UserMapper depending on how granular we want to be. Do we want to keep UserSummary alongside the actual User class or split those off as well? So that could be another five or so if we want very specific Mapper interfaces, each with maybe one or two methods, though some may only be used by other mappers rather than the wider codebase... 🤷 
Sorry if that's a bit rambly but hopefully it's enough information to make a decision.